lib/active_storage/service/open_stack_service.rb

```diff
@@ -7,7 +7,13 @@ class Service::OpenStackService < Service
     def initialize(container:, credentials:, connection_options: {})
       settings = credentials.reverse_merge(connection_options: connection_options)
 
-      @client = Fog::OpenStack::Storage.new(settings)
+      fog_openstack_storage_class = if defined?(Fog::OpenStack::Storage)
+        Fog::OpenStack::Storage
+      else
+        Fog::Storage::OpenStack
+      end
+
+      @client = fog_openstack_storage_class.new(settings)
```

Comment:
Can you apply `Fog::OpenStack::Storage` directly without the `if..else` block statement. Since this will become the standard from version 0.3 ( I think) or 1.0, it will be locked in the gemspec. And people wanting to use old versions of fog-openstack, will have bundle automatically select an old release of `activestorage-openstack`

For old versions of this gem, they will be locked. and if there is a security fix or patch, i'd release a patch version and backport it in the older versions ( which is unlikely to happen, but is achievable and could be done)